packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.routing.js

```diff
@@ -29,6 +31,15 @@ export default /* @ngInject */ ($stateProvider) => {
           (option1, option2) =>
             getOptionPrice(option1)?.value - getOptionPrice(option2)?.value,
         ),
+      renewPeriod: /* @ngInject */ (upgradeOptions, getOptionInterval) => {
+        let interval = DEFAULT_INTERVAL;
+        upgradeOptions.forEach((option) => {
+          if (getOptionInterval(option) !== DEFAULT_INTERVAL) {
+            interval = getOptionInterval(option);
+          }
+        });
+        return interval;
```

Comment:
const upgradeOption = upgradeOptions.find((option) => getOptionInterval(option) !== DEFAULT_INTERVAL)

return upgradeOption || DEFAULT_INTERVAL;

Reply:
It's not the same. I just want the interval value and not an object with all values.

Reply:
For me we have a no deterministic search.
the loop will iterate and the interval will be set more than one times a think.
So we will take every time the last value that is different from `default`, it's expected ?

Reply:
As seen together. I applied your method.